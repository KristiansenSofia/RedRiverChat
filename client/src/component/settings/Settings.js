import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'

// Import NPM-modules
import { CircularProgress } from 'material-ui/Progress'

// Import styles. settingsStyles for all imported components with a style attribute and CSS-file for classNames and id.
// import {settingsStyles} from '../../styles/SettingsStyles'
import '../../styles/Styles.css'

// Import pages for all different user settings.
import User from './user/SettingsUser'
import SuperUser from './superUser/SettingsSuperUser'
import Admin from './admin/SettingsAdmin'

/**
 *  Settings-component. Start page for settings. Users role is included in props from App.js and the
 *  user is redirected to settings based on role.
 *
 *  @author Jimmy
 */

class Settings extends Component {
  constructor (props) {
    super(props)

    this.state = {
      userRole: this.props.state.userRole
    }

    this.toggleUser = this.toggleUser.bind(this)
  }

  /**
     *  Check users role and redirect to settings based on role.
     *
     *  @author Jimmy
     */

  toggleUser () {
    switch (this.state.userRole) {
      case 'SuperUser':
        return <SuperUser state={this.props.state} />

      case 'Admin':
        return <Admin state={this.props.state} />

      case 'User':
        return <User state={this.props.state} />

      default:
        return <User state={this.props.state} />
    }
  }
  render () {
    if (this.props.state.isSignedIn === false) {
      return <Redirect to='/login' />
    }

    return (
      <div className='Settings'>
        {this.state.userRole ? (
          <div>
            {this.toggleUser()}
          </div>
        ) : (
          <CircularProgress />
        )}
      </div>

    )
  }
}

export default Settings
