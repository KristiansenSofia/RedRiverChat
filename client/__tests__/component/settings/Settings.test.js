import React from 'react';
import Settings from '../../../src/component/settings/Settings';
import renderer from 'react-test-renderer';

test('Settings renders without crashing', () => {
  const tree = renderer.create(
    <Settings state = {{ isSignedIn: true, userRole: 'User', userInfo : { username: 'username' }}}/>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
