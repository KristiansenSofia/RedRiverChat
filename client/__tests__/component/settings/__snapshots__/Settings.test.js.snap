// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders without crashing 1`] = `
<div
  className="Settings"
>
  <div>
    <div
      className="SettingsUser"
    >
      <h1
        className="MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-alignLeft-13"
        style={
          Object {
            "marginLeft": 30,
          }
        }
      >
        Inställningar
      </h1>
      <div
        className="MuiPaper-root-28 MuiPaper-elevation1-31 MuiExpansionPanel-root-25"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-63 MuiExpansionPanelSummary-root-55"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-59"
          >
            <p
              className="MuiTypography-root-1 MuiTypography-body1-10"
            >
              Användaruppgifter
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-63 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-61"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-71"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-72"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-78"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-85"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-87"
          >
            <div
              className="MuiCollapse-wrapperInner-88"
            >
              <div
                className="MuiExpansionPanelDetails-root-89"
              >
                <div
                  className="UserDetails"
                >
                  <h1
                    className="MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-alignLeft-13"
                    style={
                      Object {
                        "fontSize": "90%",
                        "marginLeft": 30,
                      }
                    }
                  >
                    Användaruppgifter
                  </h1>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root-28 MuiPaper-elevation1-31 MuiExpansionPanel-root-25"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-63 MuiExpansionPanelSummary-root-55"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-59"
          >
            <p
              className="MuiTypography-root-1 MuiTypography-body1-10"
            >
              Ändra lösenord
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-63 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-61"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-71"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-72"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-78"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-85"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-87"
          >
            <div
              className="MuiCollapse-wrapperInner-88"
            >
              <div
                className="MuiExpansionPanelDetails-root-89"
              >
                <p
                  className="MuiTypography-root-1 MuiTypography-body1-10"
                >
                  Ändra lösenord
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root-28 MuiPaper-elevation1-31 MuiExpansionPanel-root-25"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-63 MuiExpansionPanelSummary-root-55"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-59"
          >
            <p
              className="MuiTypography-root-1 MuiTypography-body1-10"
            >
              Notifikationsinställningar
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-63 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-61"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-71"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-72"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-78"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-85"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-87"
          >
            <div
              className="MuiCollapse-wrapperInner-88"
            >
              <div
                className="MuiExpansionPanelDetails-root-89"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
